### DISCUSSION TITLE
Streaming File as it Uploads

### DISCUSSION THREAD

**Discussion by mshunjan** [fastapi/fastapi#2986] [Questions] [answered]
I have users uploading large compressed flat files (typically >1gb ), I chunk them once uploaded and then analyze each chunk. I would like to do this via a stream. Essentially I would write the streaming data to a file if it is a certain number of lines, as it gets uploaded, and then process those chunks as they get uploaded. This would require reading in the data as it comes. Is this possible?

**Comment by ghandic** [comment, marked as answer]
Check out the underlying starlette package for this, see https://github.com/encode/starlette/issues/849

**Comment by tiangolo** [comment]
If you still have questions @mshunjan, it would probably be better to ask in a GitHub issue, following the template that will guide you to structure the question in a way that makes it easier to understand your intention and need, to be able to help you better. And also providing a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) that I can copy-paste to replicate it.

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.

**Reply by mshunjan** [reply]
Thanks @tiangolo.
